File: backend/app/schemas/conversation.py
import uuid
from datetime import datetime

from pydantic import BaseModel, Field


# --- Requests ---


class ConversationCreate(BaseModel):
    title: str | None = Field(None, max_length=255)


class MessageCreate(BaseModel):
    content: str = Field(..., min_length=1, max_length=4000)


# --- Responses ---


class MessageResponse(BaseModel):
    id: uuid.UUID
    conversation_id: uuid.UUID
    role: str
    content: str
    metadata_: dict | None = None
    created_at: datetime

    model_config = {"from_attributes": True}


class ConversationResponse(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    title: str
    message_count: int
    created_at: datetime
    updated_at: datetime | None

    model_config = {"from_attributes": True}


class ConversationDetailResponse(ConversationResponse):
    messages: list[MessageResponse]


class ConversationListResponse(BaseModel):
    items: list[ConversationResponse]
    total: int
    offset: int
    limit: int


class MessageListResponse(BaseModel):
    items: list[MessageResponse]
    total: int
    offset: int
    limit: int

File: frontend/src/types/conversation.ts
export interface ConversationCreate {
  title?: string | null
}

export interface MessageCreate {
  content: string
}

export interface MessageResponse {
  id: string
  conversation_id: string
  role: string
  content: string
  metadata_: Record<string, unknown> | null
  created_at: string
}

export type MessageStatus = 'confirmed' | 'pending' | 'error'

export type ChatMessage = MessageResponse & {
  status: MessageStatus
  errorReason?: string
}

export interface SendMessageVariables {
  conversationId: string
  content: string
  retryId?: string
}

export interface ConversationResponse {
  id: string
  user_id: string
  title: string
  message_count: number
  created_at: string
  updated_at: string | null
}

export interface ConversationDetailResponse extends ConversationResponse {
  messages: MessageResponse[]
}

export interface ConversationListResponse {
  items: ConversationResponse[]
  total: number
  offset: number
  limit: number
}

export interface MessageListResponse {
  items: MessageResponse[]
  total: number
  offset: number
  limit: number
}
